people_tool/static/people_tool/js/controllers/SearchController.js

```diff
@@ -0,0 +1,162 @@
+            // searchType is an element of the $scope.searchTypeOptions array
+            $scope.searchType = searchType;
+        };
+
+        // configure the datatable
+        $scope.dtInstance = null;
+        $scope.dtOptions = {
+            ajax: function(data, callback, settings) {
+                var url = djangoUrl.reverse('icommons_rest_api_proxy',
+                                            ['api/course/v2/people/']);
+                var queryParams = {
+                    offset: data.start,
+                    limit: data.length,
+                    ordering: (data.order[0].dir === 'desc' ? '-' : '')
+                              + $scope.sortKeyByColumnId[data.order[0].column]
+                };
+                queryParams[$scope.searchType.key] = $scope.queryString;
+
+                $scope.toggleOperationInProgress(true);
+
+                // todo: abort existing request?
+                $.ajax({
+                    url: url,
+                    method: 'GET',
+                    data: queryParams,
+                    dataSrc: 'data',
+                    dataType: 'json'
+                }).done(function dataTableGetDone(data, textStatus, jqXHR) {
+                    $scope.messages = [];
+                    callback({
+                        recordsTotal: data.count,
+                        recordsFiltered: data.count,
+                        data: data.results
+                    });
+                })
+                .fail(function dataTableGetFail(data, textStatus, errorThrown) {
+                    $log.error('Error getting data from ' + url + ': '
+                               + textStatus + ', ' + errorThrown);
+                    // no need for multiple messages at the moment, just
+                    // override any existing message
+                    $scope.messages = [{
+                        type: 'danger',
+                        text: 'Server error. Search cannot be completed at ' +
+                              'this time. '
+                    }];
+                    callback({
+                        recordsTotal: 0,
+                        recordsFiltered: 0,
+                        data: []
+                    });
+                })
+                .always(function dataTableGetAlways() {
+                    $scope.toggleOperationInProgress(false);
+                    $scope.$digest();
+                });
+            },
+            createdRow: function( row, data, dataIndex ) {
+                // to use our 'badge' directive within the rendered datatable,
+                // we have to compile those elements ourselves.
+                $compile(angular.element(row).contents())($scope);
+            },
+            deferLoading: 0,
+            language: {
+                emptyTable: 'There are no people to display.',
+                info: 'Showing _START_ to _END_ of _TOTAL_ people',
+                infoEmpty: 'Showing 0 to 0 of 0 people',
```

Comment:
Very Minor: Could we put a different message here? Or not even show anything as the table already displays 'There are no people to display'.